lib/src/main/kotlin/com/swmansion/starknet/data/types/transactions/TransactionReceipt.kt

```diff
@@ -18,18 +18,28 @@ sealed class TransactionReceipt {
     abstract val actualFee: Felt
     abstract val isPending: Boolean
     abstract val type: TransactionReceiptType
+
+    abstract val blockHash: Felt?
+    abstract val blockNumber: Int?
+    abstract val status: TransactionStatus?
```

Comment:
Why not add these to the `ProcessedTransactionReceipt`? Having these here defeats the purpose of even having separate `TransactionReceipt` and `ProcessedTransactionReceipt`.

Reply:
ProcessedTransactionReceipt has TransactionStatus that can't be null. Since all those fields can be null it is easier for users to use them without casting in the first place.

Reply:
Is it possible for the status to be null though? I think if you can query gateway/rpc for the TX, it'll at least have a status.

Reply:
What if tx has just been sent?

Reply:
On gateway it is possible for tx to have `NOT_RECEIVED` status (actually any invalid hash will also return that status), and for full node it'd be impossible to query for transactions that hasn't been somehow processed by StarkNet so an error will most likely be raised.

Reply:
So I guess this should be non nullable type at the root of the tx receipts type?

Reply:
I think so, we can add some default value just to be sure, I wrote about this in the answer below.